### DISCUSSION TITLE
standalone smartwatch with xdrip and sim card

### DISCUSSION THREAD

**Discussion by post2023** [NightscoutFoundation/xDrip#2819] [Q&A]
Hallo @jamorham 
I am looking for a standalone smartwatch solution for my 6 year old son. He wears the insulin pump from Ypsomed with the sensor Dexcom G6 and the software mylife CamAPS FX.

We want our son to be able to take off the insulin pump and mobile phone that runs CamAPS FX when he does sports and he only has to wear a smartwatch to monitor his blood sugar. The smartwatch would have a SIM card and would therefore have an internet connection. Our son's smartwatch could thus transmit the data to our parents' mobile phone (follower function) via the Internet and we could track his blood sugar levels remotely.

I thought of the following solution:
The smartwatch should have Bluetooth and a SIM card. Full Android or Wear OS Powered by Samsung OS would then be installed on the smartwatch and the xDrip software. The watch would pass on the values for the followers via the SIM card. For this reason, a mobile phone that would be nearby would no longer be necessary. The smartwatch would communicate with the Dexcom sensor over the non-phone channel (medical device channel) as the phone slot of the sensor is used by the Natel for the CamAPS FX software.
(as seen on the picture and my questions at next chat).

**Comment by post2023** [comment]
Now to my questions:
1.	Which xdrip+ version do I need to download that I can run this in engineering mode?
2.	Where can I download this xdrip+ version with engineering mode?
3.	Which operating system must be installed on the smartwacht for the xdrip+ version to work in engineering mode?
4.	Is it possible to install xdrip+ on the operating system full Anroid? If so which version?
5.	Can xdrip+ be installed on the Wear OS operating system? If so which version?
6.	Can I install xdrip+ on the Wear OS Powered by Samsung operating system? If yes which version?
7.	Can you recommend me a Smartwatsch with Bluetooth and SIM card?
8.	On which channel is xdrip+ set by default? On the phone channel or on the medical channel?
9.	In my case the telephone channel is used for communication between the Dexcom 6 transmitter and the cell phone. Would I now have to set xdrip+ on the smartwatch to slot 1?
10.	Is slot 1 the medical channel (as used by Bluejay GTS)? The Dxcom G6 is still supposed to work via the mobile phone by default? Otherwise the camAPS FX will not work.
11.	Can you also set this medical channel on a smartwatch?
12.	Does the medical channel also communicate via Bluetooth?
13.	Which Bluetooth version is required?
14.	Can the smartwatch installed with xdrip transmit the data directly to the parent's follower phone? Or does it still require nightscout?
15.	Do you think the drawing I showed below would work with a smartwatch? the blue part.

![image](https://user-images.githubusercontent.com/132813241/236670351-f78a9819-b5e2-4dad-baf2-56a0ae68cdca.png)
 
**Number 1: Smartwatch**
•	The smartwatch has the OS full Android or Wear OS Powred by Samsung with xDrip+ installed as master.
•	Smartwatch connected with Bluetooth via device channel with Dexcom G6 sensor.
•	Smartwatch passes the data to the parent Natel configured as follower via the mobile network with a SIM card.

**Number 2: Parent cell phone with xdrip+ as follower**
•	On the parent phone xdrip+ is installed as follower and receives the data from the smartwatch.

**Number 3: Dexcom G6 sensor**
•	The sensor is permanently connected to the phone via the phone channel on which cam APS FX is installed and connected to the smartwatch via the device channel.
•	Does the sensor have only phone channel and device channel?
•	Is it possible to configure on the smartwatch in xDrip+ that the smartwatch communicates via the device channel? According to me, yes the Dexcom does not have two phone channels.

**Number 4: Android Natel on which cam APS FX is installed.**

Thank you for your support. We want to find the best solution for our son.
Kind regards,

**Reply by Navid200** [reply]
These are the items in your list I can answer:  
1- The current stable version of xDrip, or any one after that, can use [Engineering mode](https://navid200.github.io/xDrip/docs/Engineering-Mode.html).   
2- The only place you should download xDrip from is this repository.  More detail: https://navid200.github.io/xDrip/docs/Download-xDrip.html  
3- Android if the watch is full Android.  
4- The current stable version of xDrip or any version after that can be installed on any Android 5 or newer version of Android on any device that runs Android.  
5- WearOS is not full Android.  

8- xDrip uses channel 2 by default: https://navid200.github.io/xDrip/docs/G6_slot.html  
9- If channel 2 is taken, and you want xDrip to be a collector and not a follower, you have no choice but to set xDrip to use channel 1.  
10- G6 can be controlled on either one of its two channels.  Only one device can pair to either channel.  You could pair two devices simultaneously using the two channels.  

12- Both channels use Bluetooth.  
13- xDrip supports Android version 5 and newer with the Bluetooth version included.  
14- Any Android device with xDrip on it can use one of two (open source) methods for sharing.  Those being xDrip sync or Nightscout.  https://navid200.github.io/xDrip/docs/Follow_page.html .  Neither sends data "directly" to parent's phones.  
   
You can use  BlueJay watch.  https://bluejay.website/  
However, it does not have a sim card.  Therefore, it cannot share except using Bluetooth, which is only for very close range.  So, you can set it up to collect independently from channel 1 and share with your son's phone.
Then, parents can  follow shared by the phone while your son is close to his phone.
For the periods he is in the field far from his phone, he himself can see alerts on his BlueJay watch, but, parents will not receive any readings.

**Reply by post2023** [reply]
Hallo Navid 
many thanks for your response.
Can I run the xdrip+ software in standalone engineering mode without a cell phone on the Wear OS Powered by Samsung or Wear OS operating system? 
or does xdrip only work on the full Android operating system?
many thanks for your response.
LG

**Reply by Navid200** [reply]
WearOS is not an independent stand-alone Android OS.  

@psonnera
Would you please state what WearOS is and what it can or cannot do?

**Reply by psonnera** [reply]
@Navid200 that would need to be elaborated more but Wear OS is the smartwatch version of Android as an OS. Since it's been designed for smartwatches and not phones it is a very light version of Android and doesn't include all the features you'll find on a phone version. As for what it can do and cannot, this is really application specific and depends on the features requested by the applications. In the normal environment Wear OS has a play store entry with dedicated apps, in our case Wear OS has its own app embedded in xDrip+, but it's not the full xDrip+ version. Basically some apps can be sideloaded to Wear OS and might work. xDrip+ doesn't: only the Wear extension does. Do not confuse Wear OS watches with full Android watches that are smartphones with the shape of a smartwatch. Those are not Android OS but full Android. You're right the subject deserves more documentation.

**Reply by post2023** [reply]
@psonnera 

Hallo Patick

Thank you very much for your explanation.
I want to install xdrip+ software on a smartwatch with sim card. The smartwatch should work as a master and receive the blood glucose values of the Dexcom G6 via the medical channel. For this reason, xdrip+ must support engineering mode on the smartwatch. Since the phone channel of the G6 is already used. So it is important that no phone has to be near. The smartwatch as a master should then be able to pass on the blood glucose data to a follower cell phone of the parents via the mobile radio network.
 
![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/00da9df0-4d24-451c-8217-f529605ff82e)


**Now my question is:**
•	Does the full xDrip+ version have to be installed for the solution described above?
•	Which OS must be installed on the smartwatch for the above solution to work?
•	Or can I buy a watch for this solution that has the operating system WearOS/WearOS for Samsung installed?
•	Can you recommend me a watch that would work for my solution and is waterproof?

Thank you for your support. We want to find the best solution for our son.
Kind regards,

**Reply by psonnera** [reply]
@post2023 You need a full Android watch since the Wear extension of xDrip+ is not fully functional with the G6.
Android 6 and above would be better.
No: it needs to be full Android.
You should browse AliExpress for Kospet, Lemfo, ... they are huge and the battery lasts 8 hours maximum with a SIM.

**Reply by post2023** [reply]
Hallo @Navid200 and @psonnera 

Many thanks for your help and support. I am very happy.

- Is there a step-by-step guide if I want to run xdrip+ in engineering mode and not connect to the transmitter over the phone channel?

- Very important for me not to disturb the sensor session over the phone. What do I need to do when configuring to be able to connect via the device channel (medical channel)? How do I do this?

- It is very important that my son does not disturb the connection between the sensor and the mobile phone.

Many thanks for your help. I am so grateful.

 wishes

**Reply by psonnera** [reply]
@post2023 see here https://navid200.github.io/xDrip/docs/G6_slot.html

**Reply by Navid200** [reply]
If something is in engineering mode, it means the developers do not feel confident about it always behaving the same way.
It is a very BAD idea to use engineering mode for regular operation.

**Reply by post2023** [reply]
@Navid200 
Thanks for your answer. I only need the engineering mod to set the Natel channel to the device channel (medical channel).
https://navid200.github.io/xDrip/docs/G6_slot.html

Is it then not possible to close the engineering mod again? Doesn't the channel then stay fixed on Chanel 1?

**Reply by post2023** [reply]
@psonnera 
Thanks for your answer. Sorry for the questions.

My Dexcom G6 sensor and transmitter is already up and running and is already connected to the cellular phone on which Camaps FX is connected via the cellular channel. I must not interrupt this connection and I must not stop the session.
For this reason, I have the question, how do I connect the Full Android Smartwatch to the Dexcom G6 sensor via the Devices channel (Medical Channel) without starting a new session or stopping the existing one? Is there a guide?

Do I have to apply the following setting for this solution as in this guide?
https://xdrip.readthedocs.io/en/latest/install/g6/

Thanks for your help.

**Reply by Navid200** [reply]
If you disable engineering mode, the ability to change the Bluetooth slot will be disabled.
This is because we are not sure this feature is reliable.

**Reply by post2023** [reply]
@Navid200 
Thanks for your answer. Sorry for the questions.

My Dexcom G6 sensor and transmitter is already up and running and is already connected to the cellular phone on which Camaps FX is connected via the cellular channel. I must not interrupt this connection and I must not stop the session.
For this reason, I have the question, how do I connect the Full Android Smartwatch to the Dexcom G6 sensor via the Devices channel (Medical Channel) without starting a new session or stopping the existing one? Is there a guide?

Do I have to apply the following setting for this solution as in this guide?
https://xdrip.readthedocs.io/en/latest/install/g6/

Thanks for your help.

**Reply by Navid200** [reply]
If you have a session in progress, you can connect to it without stopping.

https://navid200.github.io/xDrip/docs/Session-in-progress.html

**Reply by psonnera** [reply]
I've read that setting the watch to "Only use Wear" forces medical channel but haven't checked the code for this. Unfortunately I'm not convinced you can make the switch without having the phone connected to the sensor first.

**Reply by post2023** [reply]
Hallo @psonnera and @Navid200 

Thank you very much for your patience and help! I am very glad.  I hope I am not annoying. But I would like to try a solution if this could work for us. Once again a big thank you.

I describe once how I would like to proceed and would still have question about it. As I said, this is an attempt for me.

**myllife Loop from YpsoMed**
I have connected the sensor and the transmitter to the phone via the **phone channel (channel 2)**. This setting must not be disturbed. This technical solution is required by the myllife Loop from YpsoMed. See drawing:

![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/f2d2e79e-6eb8-4b6f-b0f8-537eaadb7918)

**Solution for sports for my son with a full Android smartwatch:** 

As a next step I want to pair the smartwatch running full Android and Xdrip+ with the sensor and transmitter. Unfortunately, I do not know how here, because I am not allowed to stop and restart the sensor. It is important here that I must make the connection via the Chanel 1 with Xdrip+ installed on the smartwatch.

![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/c4769314-d70a-454b-a3cf-466a1f3435d7)

**I have made the following settings in xdrip+:** 

**1.  data source I have selected G5/G6**

![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/32ed539c-5ef9-44f2-84d0-13a6ab0f7bf5)

Set the G6 Bluetooth slot of Chanel 2 to 1 in engineering mode. After setting the engineering mode again deactivated.
![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/e8e096fb-7aa6-4804-aa06-8019d541df82)


**=> Are these settings ok? And are these all I have to do so that the connection to the sensor and transmitter via Chanel 1?**


**Now to my questions:** 

unfortunately, I do not understand it yet. Sorry

1.  how do I pair the smartwatch (Chanel 1) with the sensor if it is already in operation?
2.  can I simply enter the Dxcom Tranmitter ID under setting? Or will this restart the sensor? This must not happen.

![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/353c5c91-8792-41d9-b198-cbe3a82c2e39)

Sorry that I do not understand how I can pair the sensor without restarting in (warm-up phase). Can you explain this to me again. Thank you for your patience.

many greetings

**Reply by psonnera** [reply]
Starting an already started sensor won't stop it.
Stopping it will.
If you don't ask xDrip+ to stop the sensor it won't do it.
Does it make sense?

**Reply by Navid200** [reply]
To pair a transmitter to xDrip, you don't need to start a sensor as explained here: https://navid200.github.io/xDrip/docs/Session-in-progress.html

If you disable engineering mode, xDrip will automatically switch back to slot 2, as explained here https://navid200.github.io/xDrip/docs/G6_slot.html, which could interfere with CamAPS.

**Reply by post2023** [reply]
Hallo @Navid200 and @psonnera 

Thank you very much for the answer. 

I still don't understand it completely. Sorry. This tutorial doesn't say how I can make the connection. I do not understand it. Sorry. 
If I understand the instructions correctly, I am not allowed to perform the steps I list below, right? 
So I would have to turn off the phone on which the software Camaps FX runs on it? 
How does the BlueJay watch make this connection. There it also works today that both runs without problems. 
https://navid200.github.io/xDrip/docs/Session-in-progress.html

I have briefly written down the steps as I understood it. I would appreciate if you could take a quick look at this and tell me if I understood it correctly? If I work through these steps the same way, the sensor and transmitter will not be restarted by xdrip+. I can still start and stop the sensor and transmitter with the camapx FX app? There is no warm up phase started by xdrip+?

![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/e7a93483-8039-41e9-bf66-f9ecff1177a2)
![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/6f770ec4-0dc2-4d90-95f0-71cec1d04bdc)
![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/0f43a991-b01e-4183-b6a3-f0008e05b989)
![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/31bc8229-b9f4-438e-91b9-a129deb810de)

**Reply by Navid200** [reply]
When you tap on start sensor, you start a session.  In case of G6, the session is on the transmitter.  It doesn't matter if you want to use xDrip or CamAPS or t:slim pump or BlueJay or whatever.  After you have started a session, you don't need to start again in xDrip.

When you establish connectivity to a transmitter that has a session in progress, xDrip will show the readings.
If you have established connectivity and you see no errors on the status page and if last connected shown on status page is consistently less than 5 minutes, it means you have connectivity.  If after that, you still have no readings, there is nothing wrong tapping on start sensor.  xDrip will ask you for the calibration code.  If there is a session in progress, the calibration code you enter makes no difference.  You can enter the same calibration code you used when you originally started the session.

You are focusing on the wring thing here.
I understand you want to go over everything and after you understand everything and only then, go ahead and execute.
But, you are focusing on the wrong hurdle.

Your main hurdle is that a full Android watch is very big and heavy for a small child to wear while participating in a sport.
If I were you, I would focus on the hurdles.
I have tried from the beginning bringing to your attention what your main hurdles are.
But, you keep going back to your diagrams.

It is admirable how methodical you work.  I appreciate that.  it is a good thing to plan before you execute.  I hate it when someone just jumps in and then asks  questions.
But, you seem to be at the other end of the spectrum.

How is your son going to wear a full Android watch?  Have you found one that is light and small enough?

**Reply by post2023** [reply]
@Navid200 

Thank you very much for your help. I am really glad. Will test it tomorrow with a Samsung Natel and hope it works. 

Your input regarding smartwatch I have taken to heart and have in the meantime searched for a watch and also for solutions as my son can wear the watch during sports. 

**Old solution with Natel:** 
Currently, he carries the Natel (Samsung S22) in a fanny pack with him during sports to erhlaten the values of the Dexcoms. The cell phone has the dimensions 150mm x 75mm x 12mm. 


**New solution with smartwatch:** 
He will wear the smartwatch on his upper arm and put on a sweatband to protect it. 


The following smartwatch are available: 
Lokmat Appllp 2 pro
[https://de.aliexpress.com/item/1005004561819226.html?gatewayAdapt=glo2deu](url)

Lemfo Lem16 Link => this tip gave me psonnera
[https://de.aliexpress.com/i/1005004547246932.html?gatewayAdapt=glo2deu](url)

The watches have the dimensions 51 x 51 x 16 mm. So this solution is much smaller and handier than the solution with the natel. It can also be carried in the fanny pack. 

What do you think about it?

**Reply by Navid200** [reply]
I never thought of wearing the watch up on the arm.  OK, that can be done.
Where will the G6 sensor be?  Hopefully not on the other arm.

**Reply by post2023** [reply]
Since our son has little fat on the arm, he wears the sensor on the buttocks

**Reply by Navid200** [reply]
OK,

When I first got my BlueJay, I was surprised that it had no reception if I wore it on the wrist  and the sensor on the other arm.
The only way I can explain this is that my body is in the middle.  Our body is mostly water.  Bluetooth range is significantly reduced in water.

So, hopefully, you can test this to make sure it works and hopefully you can return it if it doesn't.

**Reply by post2023** [reply]
@Navid200 
my son currently also has the BlueJay and it works.

For this reason I would like to test it tomorrow with my Samsung Natel and hope it works so. 

Wish me luck for tomorrow. :o)

My biggest wish is to support my 6 year old son in the best possible way and for this no effort is too much. 

I hope very much that I can build him a solution that is best for this time. Unfortunately, he does not yet feel the hypos during sports and play so well. This will certainly improve later and then he may no longer need such reductions. It is also important that he develops a feeling for his body as quickly as possible.

**Reply by post2023** [reply]
@Navid200 

Hallo
Sorry still have a quick question. Android 11 is supported by xdrip+ without any problems? 

Thank you very much.

**Reply by psonnera** [reply]
yes

**Reply by Navid200** [reply]
As well as 12 and 13

**Reply by Navid200** [reply]
The xDrip limitation in that respect is that it doesn't support Android 4 any longer.

Android 5 will be retired soon.

There is no way we can test all devices specially when a new Android is released.
We rely on reports from users.

Android 11 is quite mature now.

**Reply by post2023** [reply]
@Navid200 

Hello 

I have now done a quick test one with two Android Natel. Sees picture

**Initial situation:** 
Phone 2 with CampAPS FX was already connected to the Dexcom G6 sensor on slot 2. Next, I paired the Phone 1 with xDrip+ to the Dexcom G6 via slot 1. I received readings after about 10 minutes on Phone 1. Readings continued to be displayed on Phone two. 

**Questions:** 

- If on Phone 1 and Phone 2 the values are displayed at the same time does it work? Or?

- If I now on the Phone 1 on the xDrip+ running in the menu on sensor stop presses then the sensor would be stopped and I would also have no more values on the Phone 2 with CamAPS FX? I don't quite understand this yet with xDrip+. 

- Who is now the master for the sensor to stop? xDrip+ or CAmAPS FX? Until now it was CamAPSFX.

- At what interval in minutes does xDrip+ poll the sensor for blood glucose readings? Is it possible to set this? 

![image](https://github.com/NightscoutFoundation/xDrip/assets/132813241/d3de5f4b-99d3-4619-883c-be30baa6f9e5)

Thanks again. I think so far it is coming fine. What do you think?

**Reply by Navid200** [reply]
The transmitter has two slots.  It is paired to two different devices.  Both of those devices can show the readings simultaneously.

The master is the transmitter.  If you tell the transmitter to stop, it will stop the session.  You should never stop sensor with G6 except when you intend to end the session.

xDrip does not define the interval.  It is the transmitter that wakes up once every 5 minutes and transmits for a few seconds.  xDrip sets a timer and wakes up at the same time and starts listening for the next reading.

**Comment by pdg212** [comment]
> Hallo @jamorham I am looking for a standalone smartwatch solution for my 6 year old son. He wears the insulin pump from Ypsomed with the sensor Dexcom G6 and the software mylife CamAPS FX.
> 
> We want our son to be able to take off the insulin pump and mobile phone that runs CamAPS FX when he does sports and he only has to wear a smartwatch to monitor his blood sugar. The smartwatch would have a SIM card and would therefore have an internet connection. Our son's smartwatch could thus transmit the data to our parents' mobile phone (follower function) via the Internet and we could track his blood sugar levels remotely.
> 
> I thought of the following solution: The smartwatch should have Bluetooth and a SIM card. Full Android or Wear OS Powered by Samsung OS would then be installed on the smartwatch and the xDrip software. The watch would pass on the values for the followers via the SIM card. For this reason, a mobile phone that would be nearby would no longer be necessary. The smartwatch would communicate with the Dexcom sensor over the non-phone channel (medical device channel) as the phone slot of the sensor is used by the Natel for the CamAPS FX software. (as seen on the picture and my questions at next chat).



Hello, I am wanting to do the exact same thing - just wondered if you managed to get this working?
